### DISCUSSION TITLE
Adding preset prefix option for preset-mini

### DISCUSSION THREAD

**Discussion by chu121su12** [unocss/unocss#303] [Ideas]
@antfu
1. For easier integration with other framework, what's your take on adding prefix option on the class-based preset? Attributify  already support prefix.
2. How about configurable variable prefix (`--un-`)?

**Comment by antfu** [comment]
Go for it. But probably `un-`? since double dashes are required for CSS vars.

**Comment by chu121su12** [comment]
The second question is more about configuring the css-variable prefix, instead of the selector, in some of the rules:

Something like:

```js
return {
  [`--un-${varName}-opacity`]: 1,
  [property]: `rgba(${rgba.slice(0, 3).join(',')},var(--un-${varName}-opacity))`,
}
```
to 
```js
config.cssVarPrefix = 'un'
return {
  [`--${config.cssVarPrefix}-${varName}-opacity`]: 1,
  [property]: `rgba(${rgba.slice(0, 3).join(',')},var(--${config.cssVarPrefix}-${varName}-opacity))`,
}
```

**Comment by antfu** [comment]
Ah, I didn't see there are two questions,

1. We already have it https://github.com/antfu/unocss#utilities-preprocess--prefixing
2. I prefer:
```ts
config.cssVarPrefix = 'un-'
return {
  [`--${config.cssVarPrefix}${varName}-opacity`]: 1,
  [property]: `rgba(${rgba.slice(0, 3).join(',')},var(--${config.cssVarPrefix}${varName}-opacity))`,
}
```
